### PR TITLE
syntax highlighting updates for vyper v0.4.0

### PR DESCRIPTION
**Pull request by charles-cooper**

### CODE REVIEW

File: syntax/vyper.vim [hunk 1]
```diff
@@ -12,16 +12,15 @@ endif
 "Keywords
 syn keyword vyperNatspec @title @licence @author @notice @dev @param @return @version
```
Comment (pcaversaccio) [inline syntax/vyper.vim:13]: should be `@license` and also `@custom:...` is missing

File: syntax/vyper.vim
Comment (pcaversaccio) [inline syntax/vyper.vim:1]: The `unsafe_` math functions are missing and some further builtins like `print`, `_abi_encode`, `_abi_decode`, `blobhash`, `blockhash`, `isqrt`, `sqrt`, `min_value`, `max_value`, `epsilon`, `abs`, `uint2str`, `send`, `selfdestruct`, `raw_revert`, `raw_log`, `create_from_blueprint`, `create_copy_of`, `create_minimal_proxy_to`  or the type `immutable`.

File: syntax/vyper.vim [hunk 2]
```diff
@@ -2,7 +2,7 @@
 " Language:             Vyper (https://github.com/vyperlang/vyper)
 " Author:               chgue (https://github.com/chgue)
 " URL:                  https://github.com/vyperlang/vim-vyper
-" Last Change:          2020-10-10
+" Last Change:          2024-06-07
 " Filenames:            *.vy
```
Comment (pcaversaccio) [inline syntax/vyper.vim]: can we have also `.vyi` here?

Comment (charles-cooper) [reply syntax/vyper.vim]: i think that's already handled in ftdetect actually

File: syntax/vyper.vim
Comment (charles-cooper) [reply syntax/vyper.vim:1]: meh, it's a redundant rule anyway, they are already all highlighted the same as `vyperFunction`

File: syntax/vyper.vim (hunk 1 above)
Comment (charles-cooper) [reply syntax/vyper.vim:13]: i think those are already handled by the `syn match vyperNatspec` rule

Comment (pcaversaccio) [reply syntax/vyper.vim:13]: What I refer to actually: image

**Review by pcaversaccio — APPROVED** [review]
Let's get this merged and adjust later if something is missing.

**Pull request merged**